docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,308 @@
+
+### How do we distribute rewards?
+Currently, rewards are distributed as follows: The consumer [periodically sends rewards](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/consumer/keeper/distribution.go#L148) on the provider `ConsumerRewardsPool` address.
+The provider then [transfers those rewards to the fee collector address](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/keeper/distribution.go#L77) and those transferred rewards are distributed to validators and delegators.
+
+In PSS, we distribute rewards only to validators that actually validate the consumer chain. To do this, we have a pool associated with each consumer chain and consumers IBC transfer the rewards to this pool.
+Then, because we do not use the canonical fee pool and hence rewards are not automatically distributed, we will create a new and modified version of [AllocateTokens](https://github.com/cosmos/cosmos-sdk/blob/v0.47.7/x/distribution/keeper/allocation.go#L14) as follows:
+
+```
+func (k Keeper) AllocateTokensFromPool(ctx sdk.Context, poolAddress string, totalPreviousPower int64, bondedVotes []abci.VoteInfo) {
+	rewardsCollectedInt := k.bankKeeper.GetAllBalances(ctx, poolAddress)
+	// rest in the same that `AllocateTokens` operates
+	...
+```
+
+In [TransferRewardsToFeeCollector](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/keeper/distribution.go#L61) we then do:
+
+```
+for _, chain := range consumerChains {
+    ...
+    totalPower := 0
+    var []abci.VoteInfo votes
+    for _, val := range (validators that opted in or Top N) {
+        if val has opted in for more than X blocks {
+            votes = append(votes, []abci.VoteInfo{{Validator: val, SignedLastBlock: true}})
+        }
+    }
+    
+    k.AllocateTokens(ctx, pool associated with this chain, totalPower, votes)
+    ...
+}
+```
+
+instead of calling:
+```
+// 4. Transfer the balance to the fee collector address
+err := k.bankKeeper.SendCoinsFromModuleToModule(
+    ctx,
+    types.ConsumerRewardsPool,
+    k.feeCollectorName,
+    sdk.NewCoins(balance),
+)
+```
+
+Note that we would only distribute rewards to validators that are opted in but are opted in for some time (e.g., 10000 blocks) to avoid cases where validators opt in and out in short intervals just in order to receive rewards.
+
+### Misbehaviour
+
+#### Fraud votes
+For fraud votes, we add a new type `FraudVoteProposal` similar to [EquivocationProposal](https://github.com/cosmos/interchain-security/pull/703). The time to detect incorrect execution evidence is 7 days (the same time as that of detecting a light client attack on a client with a `trustingPeriod` of 14 days); We have 7 days to detect an incorrect execution and then 14 days for the proposal to be accepted.
```

Comment:
This numbers depend on the UnbondigPeriods on both provider and consumers.

Reply:
Resolving this because we do not talk about fraud votes anymore in this ADR.